asdf/asdf.py

```diff
@@ -867,6 +868,9 @@ def _open_asdf(
                 self._blocks.read_internal_blocks(fd, past_magic=True, validate_checksums=validate_checksums)
                 self._blocks.read_block_index(fd, self)
 
+            if tree_mapper is not None:
```

Comment:
If I'm following this correctly, `tree_mapper` is only used to transform the tagged tree on read. Is that correct? If so, can we call it `tagged_tree_transform`?